lib/complex/compile_time_checks.ex

```diff
@@ -0,0 +1,23 @@
+defmodule Complex.CompileTimeChecks do
```

Comment:
Btw, I don't think we need this module. You could define them as anonymous functions inside the Complex module, given it is only ever used there. :)

Reply:
I think this ended up leaking from when I had bugs regarding the compile-time check. It's probably not needed anymore

Reply:
When I moved the function back, the compiler complained about undefined function

Reply:
It needs to be an anonymous function.

    math_defined? = fn …

defined in the module body. If necessary, check String.Unicode in the elixir repo itself!